modules/generator/processor/localblocks/query_range.go

```diff
@@ -62,15 +63,21 @@ func (p *Processor) QueryRange(ctx context.Context, req *tempopb.QueryRangeReque
 		jobErr = atomic.Error{}
 	)
 
+	maxSeries := int(req.MaxSeries)
+	totalRawResultsSeriesCount := int32(0)
+
 	if p.headBlock != nil && withinRange(p.headBlock.BlockMeta()) {
-		wg.Add(1)
-		go func(w common.WALBlock) {
-			defer wg.Done()
-			err := p.queryRangeWALBlock(ctx, w, rawEval)
-			if err != nil {
-				jobErr.Store(err)
-			}
-		}(p.headBlock)
+		if maxSeries == 0 || (syncAtomic.LoadInt32(&totalRawResultsSeriesCount) < int32(maxSeries)) {
```

Comment:
all of this code external to the evaluator to track max series feels strange to me. should we push this down into the evaluator and just ask it if max series has been exceeded? 

wdyt @mdisibio ?